packages/model-serving/src/components/deploymentWizard/EditModelDeploymentPage.tsx

```diff
@@ -1,12 +1,88 @@
 import React from 'react';
+import { useLocation, useParams } from 'react-router-dom';
+import { useExtensions } from '@openshift/dynamic-plugin-sdk';
+import { byName, ProjectsContext } from '@odh-dashboard/internal/concepts/projects/ProjectsContext';
+import { Bullseye, Spinner } from '@patternfly/react-core';
+import { ProjectKind } from '@odh-dashboard/internal/k8sTypes';
+import { setupDefaults } from '@odh-dashboard/internal/concepts/k8s/K8sNameDescriptionField/utils';
 import ModelDeploymentWizard from './ModelDeploymentWizard';
+import { ModelDeploymentWizardData } from './useDeploymentWizard';
+import { Deployment, isModelServingPlatformExtension } from '../../../extension-points';
+import {
+  ModelDeploymentsContext,
+  ModelDeploymentsProvider,
+} from '../../concepts/ModelDeploymentsContext';
+import { useProjectServingPlatform } from '../../concepts/useProjectServingPlatform';
 
-const EditModelDeploymentPage: React.FC = () => (
-  <ModelDeploymentWizard
-    title="Edit model deployment"
-    description="Update your model deployment configuration."
-    primaryButtonText="Update deployment"
-  />
-);
+const EditModelDeploymentPage: React.FC = () => {
+  const { namespace } = useParams();
+
+  const { projects, loaded: projectsLoaded } = React.useContext(ProjectsContext);
+  const currentProject = projects.find(byName(namespace)) ?? undefined;
+
+  const platformExtensions = useExtensions(isModelServingPlatformExtension);
+  const { activePlatform } = useProjectServingPlatform(currentProject, platformExtensions);
+
+  if (!projectsLoaded) {
+    return (
+      <Bullseye>
+        <Spinner />
+      </Bullseye>
+    );
+  }
+
+  if (!currentProject || !activePlatform) {
+    return null;
+  }
+
+  return (
+    <ModelDeploymentsProvider modelServingPlatforms={[activePlatform]} projects={[currentProject]}>
+      <EditModelDeploymentContent project={currentProject} />
+    </ModelDeploymentsProvider>
+  );
+};
 
 export default EditModelDeploymentPage;
+
+const EditModelDeploymentContent: React.FC<{ project: ProjectKind }> = ({ project }) => {
+  const location = useLocation();
+  const { deployments, loaded } = React.useContext(ModelDeploymentsContext);
+
+  const existingDeployment = React.useMemo(() => {
+    const pathParts = location.pathname.split('/');
+    const deploymentName = pathParts[pathParts.length - 1];
```

Comment:
`name` should be pointing to the same deployment name in the url. (Maybe we could change `name` to something more accurate though like deployment name or something
```suggestion
  const { name } = useParams();
  const { deployments, loaded } = React.useContext(ModelDeploymentsContext);

  const existingDeployment = React.useMemo(() => {
```